Documentation/Development/KeyCodeConcepts.md

```diff
@@ -0,0 +1,27 @@
+
+## Effects
+Effects are basically Behaviors which also have a concept of duration. One might build ongoing spell effects, poisons, or other temporary states such as being "stunned" as Effects.
+
+## Actions
+Actions are the commands that a player can issue. They inherit from GameAction and are adorned with a number of attributes that help expose the command to the system, and to the user through the "help" Action and the "commands" Action. Note that lots of "interactive" actions the user can perform are actually dictated through relevant Behaviors instead of global Action commands.
+
+For example, there is no global "west" nor "move" command. Instead, the command is presented in a context-sensitive way by the presence of an ExitBehavior.
+
+## Database
+We are in the process of moving away from a relational database model, to a model that is largely (or completely) driven by a document database. Eventually, this will make it very easy to import/export/share arbitrary aspects of the game world, such as entire areas, or even just single interesting objects.
+
+## Telnet
+Text-based MUDs are generally built on the Telnet protocol. This is a core aspect of WheelMUD. There are a lot of useful telnet options which will be supported out of the box, which upgrade the player's experience automatically depending on the supported feature set of their Telnet Client. There are many Telnet clients out there; many established MUD players utilize powerful clients like zMUD/cMUD, but even Windows has the "Telnet.exe" client (which can be enabled out of the box through the Programs and Features / "Turn Windows features on and off" control panel).
+
+## Rule Sets
+A rule set is what defines a particular gaming system. Server administrators should have the option of starting with a particular rule set, or building their own from scratch. The core WheelMUD team has selected "[Warrior, Rogue & Mage](http://www.stargazergames.eu/games/warrior-rogue-mage)" as the sample reference rule set, due to its simplicity and open nature. Additional rule sets may show up in the roadmap after version 1.0.
+
+## DLL Extensibility
+The WheelMUD server discovers components of the game through MEF / reflection. One could drop in the DLL for a given ruleset and remove another, and relaunch the game to get a whole new game experience. Ideally, some day we'd like to support the idea of "non-reboot upgrades" where a new DLL can be dropped in to the production server, and the server discovers it and starts using the new versions of the classes for all new interactions, without even having to reboot / kick off all the players. While this isn't a reality yet, it does demonstrate how our server is powerfully composable even at the DLL level.
```

Comment:
Are you saying we should add a ## Interfaces section describing how we want to see them used?